lib/sift/filter.rb

```diff
@@ -2,18 +2,17 @@ module Sift
   # Filter describes the way a parameter maps to a database column
   # and the type information helpful for validating input.
   class Filter
-    attr_reader :parameter, :default, :custom_validate, :scope_params, :scope_types
+    attr_reader :parameter, :default, :custom_validate
 
-    def initialize(param, type, internal_name, default, custom_validate = nil, scope_params = [], scope_types = [])
+    def self.scope_type?(type)
+      type == :scope || type.respond_to?(:key?)
```

Comment:
Do you think the `type.respond_to?(:key?)` is needed here? I think before it was mostly used for raising an error message which is now triggered by `valid_scope_params`